toqito/matrix_props/is_permutation.py

```diff
@@ -0,0 +1,71 @@
+"""Is matrix a permutation matrix."""
+import numpy as np
+
+def is_permutation(mat: np.ndarray) -> bool:
+    r"""
+    Determine if a matrix is a permutation matrix [WikSquare]_.
+
+    A matrix is a permutation matrix if each row and column has a single element of 1 and all others 0.
+
+    Examples
+    ==========
+
+    Consider the following matrix
+
+    .. math::
+        A = \begin{pmatrix}
+                1 & 2 & 3 \\
+                4 & 5 & 6 \\
+                7 & 8 & 9
+            \end{pmatrix}
+
+    our function indicates that this is indeed a square matrix.
+
+    >>> from toqito.matrix_props import is_permutation
+    >>> import numpy as np
+    >>> A = np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
+    >>> is_permutation(A)
+    True
+
+    Alternatively, the following example matrix :math:`B` defined as
+
+    .. math::
+        B = \begin{pmatrix}
+                1 & 2 & 3 \\
+                4 & 5 & 6
+                7 & 8 & 9
+            \end{pmatrix}
+
+    >>> from toqito.matrix_props import is_permutation
+    >>> import numpy as np
+    >>> B = np.array([[1, 0, 0], [1, 0, 0], [1, 0, 0]] )
+    >>> is_permutation(B)
+    False
+
+    References
+    ==========
+    .. [WikSquare] Wikipedia: Permutation matrix.
+        https://en.wikipedia.org/wiki/Permutation_matrix
+
+    :param mat: The matrix to check.
+    :return: Returns :code:`True` if the matrix is permutation and :code:`False` otherwise.
+    """
+    if len(mat.shape) != 2:
+        raise ValueError("The variable is not a 2-D matrix.")
+
+    if not is_square(mat):
+        return False
+
+    for i in mat:
+      for j in i:
+        if (j!=0)&(j!=1):

```

Comment:
For brevity's sake:
```python
if (j != 0) and (j != 1):
    return False
```

Reply:
Done, thanks. 👍  and that is so true.  I think my previous use of the & was incorrect for python after checking python syntax closer as & is a bitwise and and I wanted a logical one.  My code was previously working by chance since booleans can be a single bit and not by design.  After this change it now works by design.  Thanks!

As a side note the code checking tool afterwards made be change it to the "in" syntax instead which is easier to read and understand.